### PR TITLE
optimize relay router

### PR DESCRIPTION
**Pull request by zii**
skip if dest.Id < finger.startID

### CODE REVIEW

File: overlay/chord/relay.go [hunk 1]
```diff
@@ -52,6 +52,9 @@ func (rr *RelayRouting) GetNodeToRoute(remoteMsg *node.RemoteMessage) (*node.Loc
 		if finger.IsEmpty() {
 			continue
 		}
+		if CompareID(finger.startID, remoteMsg.Msg.DestId) > 0 {
```
Comment (yilunzhang) [inline overlay/chord/relay.go:55]: This is actually not correct, an example:

localNode Id: `0xf0`
finger: `[0xff, 0x01]`
remoteMsg DestId: `0x00`

In this example, next hop should be finger[0]

**Comment by zii** [comment]
Oh, is it necessary to compute sector overlap?

**Comment by yilunzhang** [comment]
> Oh, is it necessary to compute sector overlap?

I'm not sure what you mean exactly by sector overlap, but basically you cannot just compare id. You need to check if it's in between (which is what `BetweenIncl(rr.chord.LocalNode.Id, remoteMsg.Msg.DestId, rn.Id)` is doing in the current code).

**Comment by yilunzhang** [comment]
Closed due to inactivity. Feel free to reopen it if needed

**Pull request closed without merging**